server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/MapperEngine.java

```diff
@@ -0,0 +1,212 @@
+
+        // If there is only one possibility return it.
+        if (possibleMinorImageTags.size() == 1) {
+            tag = possibleMinorImageTags.get(0).getName();
+        } else {
+            tag = determineTagFromMinorVersion(possibleMinorImageTags, semVer);
+        }
+        return tag;
+    }
+
+    /**
+     Searches for tags with a equal mayor version
+     */
+    private List<DockerImageTag> getTagsWithEqualMayorVersion(DockerImage img, int versionNumber) {
+        //Get all tags with the same mayor version (index 0) this one is required to be equal
+        List<DockerImageTag> eqMajorVersionTags = img.getVersionableTags().stream()
+            .filter(e -> e.toVersion().getMajor() == versionNumber).collect(Collectors.toList());
+
+        //Fail if there are no tags with the same major version
+        if (eqMajorVersionTags.size() == 0) {
+            throw new UnsupportedOperationException("Could not find base image. " +
+                "No tag with major version '" + versionNumber + "' found.");
+        }
+        return eqMajorVersionTags;
+    }
+
+    /**
+     Looks for tags that have none or a greater or equal minor version
+     */
+    private List<DockerImageTag> getPossibleMinorImageTags(Semver version, List<DockerImageTag> eqMayorVersionTags) {
+        // Look for minor version candidates (has to be be greater or equal to the version number given)
+        // A check if the value exists is not needed because the array length is at least 2
+        List<DockerImageTag> possibleMinorImageTags = eqMayorVersionTags.stream()
+            .filter(e -> e.toVersion().getMinor() >= version.getMinor()).collect(Collectors.toList());
+        possibleMinorImageTags.sort(TAG_COMPARATOR_MINOR_VERSION);
+
+        // If no image fulfilling the condition is found, the mapping fails
+        if (possibleMinorImageTags.size() == 0) {
+            throw new UnsupportedOperationException("Could not find base image. " +
+                "No tag with greater or equal minor version '" + version + "' found.");
+        }
+        return possibleMinorImageTags;
+    }
+
+    /**
+     Tries to determine the tag from the list of possible minor versions
+     */
+    private String determineTagFromMinorVersion(List<DockerImageTag> possibleTags, Semver version) {
+        DockerImageTag resultTag = null;
+        int i = 0;
+        DockerImageTag tag;
+        Semver tagVersion;
+        do {
+            tag = possibleTags.get(i);
+            tagVersion = tag.toVersion();
+            if (tagVersion.getMinor() == version.getMinor()) {
+                int j = i;
+                DockerImageTag prev = null;
+                DockerImageTag t = null;
+                Semver tv;
+                do {
+                    prev = t;
+                    t = possibleTags.get(j);
+                    tv = t.toVersion();
+                    j++;
+                } while (tv.getMinor() == version.getMinor() && j < possibleTags.size());
```

Comment:
`Integer` (in contrast to `int`) cannot be compared with `==`, they are complex objects, therefore you need to use `equals`.